nixos/modules/services/security/bitwarden-directory-connector.nix

```diff
@@ -0,0 +1,339 @@
+        };
+        suffix = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Suffix for the email, normally @example.com.";
+          example = "@example.com";
+        };
+      };
+
+      users = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = lib.mdDoc "Sync users.";
+        };
+        path = mkOption {
+          type = types.str;
+          description = lib.mdDoc "User directory, relative to root.";
+          example = "ou=users";
+        };
+        objectClass = mkOption {
+          type = types.str;
+          description = lib.mdDoc "A class that users will have.";
+          example = "inetOrgPerson";
+        };
+        emailAttribute = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Attribute for a users email.";
+          example = "mail";
+        };
+        filter = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Filter for users.";
+          example = "(memberOf=cn=sales,ou=groups,dc=example,dc=com)";
+        };
+      };
+      groups = {
+        enable = mkOption {
+          type = types.bool;
+          default = false;
+          description = lib.mdDoc "Sync groups.";
+        };
+        path = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Group directory, relative to root.";
+          example = "ou=groups";
+        };
+        objectClass = mkOption {
+          type = types.str;
+          description = lib.mdDoc "A class that groups will have.";
+          example = "groupOfNames";
+        };
+        nameAttribute = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Attribute for a name of group.";
+          example = "cn";
+        };
+        filter = mkOption {
+          type = types.str;
+          description = lib.mdDoc "Filter for groups.";
+          example = "(cn=sales)";
+        };
+      };
+    };
+
+    env = {
+      description = "Env files to be passed in.";
+      ldap = mkOption rec {
+        type = types.str;
+        description = "Auth for the LDAP, has ${cfg.ldap.pw_env}";
+      };
+      bitwarden = mkOption rec {
+        type = types.str;
+        description = "Auth for Bitwarden, has BW_CLIENTID and BW_CLIENTSECRET";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    users.groups."${cfg.user}" = {};
+
+    users.users."${cfg.user}" = {
+      createHome = true;
+      isSystemUser = true;
+      home = "${cfg.directory}";
+      group = "${cfg.user}";
+      homeMode = "711";
+    };
+
+    systemd = {
+      timers."${cfg.user}" = {
+        description = "Timer for ${cfg.user}";
+        wantedBy = ["timers.target"];
+        partOf = ["${cfg.user}.service"];
+        timerConfig = {
+          OnCalendar = cfg.sync.interval;
+          Unit = "${cfg.user}.service";
+          Persistent = true;
+        };
+      };
+
+      services."${cfg.user}" = {
```

Comment:
It is a bit surprising to have the name of the service attribute depend on the configured user. Is there a particular reason for that?

Reply:
That is probally just habit now for me, I think I started doping this because I had to create a large amount of services and timers and it seemed the easiest, I presume switchign it out for the pname or ``bwdc`` would work?